openfl/component/director/director.py

```diff
@@ -0,0 +1,236 @@
+            del self.experiment_stash[caller][experiment_name]
+
+    def collaborator_health_check(self, *, collaborator_name: str,
+                                  is_experiment_running: bool,
+                                  valid_duration: int) -> bool:
+        """Accept health check from envoy."""
+        is_accepted = False
+        shard_info = self._shard_registry.get(collaborator_name)
+        if not shard_info:
+            logger.error(f'Unknown shard {collaborator_name}')
+            return is_accepted
+        is_accepted = True
+
+        shard_info['is_online']: True
+        shard_info['is_experiment_running'] = is_experiment_running
+        shard_info['valid_duration'] = valid_duration
+        shard_info['last_updated'] = time.time()
+
+        return is_accepted
+
+    def get_envoys(self) -> list:
+        """Get a status information about envoys."""
+        logger.info(f'Shard registry: {self._shard_registry}')
+        envoy_infos = []
+        for envoy in self._shard_registry.values():
+            envoy_info = director_pb2.EnvoyInfo(
+                shard_info=envoy['shard_info'],
+                is_online=time.time() < envoy['last_updated'] + envoy['valid_duration'],
+                is_experiment_running=envoy['is_experiment_running']
+            )
+            envoy_info.valid_duration.seconds = envoy['valid_duration']
+            envoy_info.last_updated.seconds = int(envoy['last_updated'])
+
+            envoy_infos.append(envoy_info)
+
+        return envoy_infos
+
+    @staticmethod
+    def create_workspace(experiment_name, npbytes):
+        """Create the aggregator workspace."""
+        if os.path.exists(experiment_name):
+            shutil.rmtree(experiment_name)
+        os.makedirs(experiment_name)
+
+        arch_name = f'{experiment_name}/{experiment_name}' + '.zip'
+        logger.info(f'arch_name: {arch_name}')
+        with open(arch_name, 'wb') as content_file:
+            content_file.write(npbytes)
+
+        shutil.unpack_archive(arch_name, experiment_name)
+
+    async def _run_aggregator(
+            self,
+            experiment_sender,
+            initial_tensor_dict,
+            experiment_name,
+            collaborator_names,
+            plan='plan/plan.yaml',
+    ):  # TODO: path params, change naming
+        """Run aggregator."""
+        cwd = os.getcwd()
+        os.chdir(f'{cwd}/{experiment_name}')
+        plan = Plan.parse(plan_config_path=Path(plan))
+
+        plan.authorized_cols = list(collaborator_names)
+
+        logger.info('🧿 Starting the Aggregator Service.')
+        aggregator_server = plan.interactive_api_get_server(
+            tensor_dict=initial_tensor_dict,
+            chain=self.root_ca,
+            certificate=self.cert,
+            private_key=self.key,
+            disable_tls=self.disable_tls,
+        )
+        self.experiment_stash[experiment_sender][experiment_name] = \
+            aggregator_server
```

Comment:
We don't sure here that `experiment_sander` and `experiment_name` are valid. Could we use `dict.get()` here?